### PR TITLE
Lab1 - ripe-banana - michelle-kevin-tom

### PR DESCRIPTION
**Pull request by mizutombo**
* Initial code submittal.

### CODE REVIEW

File: lib/models/review.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const schema = new Schema({
+    rating: {
+        type: Number,
+        rater(rateNum){
```
Comment (jluangphasy) [inline lib/models/review.js]: Mongoose has a built-in validator for `min` and `max`.

http://mongoosejs.com/docs/validation.html

File: lib/models/review.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            return rateNum > 0 && rateNum <= 5;
+        },
+        message: 'rate the film on a scale from 1-5, 5 being the high.' 
+    },
+    review: {
+        type: String,
+        reviewer(reviewMsg){
```
Comment (jluangphasy) [inline lib/models/review.js]: Mongoose has a built-in validator for `maxlength`.

http://mongoosejs.com/docs/validation.html

File: lib/models/actor.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const schema = new Schema({
+	name: {
+		type: String,
+		required: true
+	},
+	dob: Date
```
Comment (jluangphasy) [inline lib/models/actor.js]: Should make `dob` field required.

File: lib/models/film.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	},
+	studio: {
+		type: Schema.Types.ObjectId,
+		ref: 'Studio'
+	},
+	released: {
+		type: Date,
+		required: true
+	},
+	actors: {
```
Comment (jluangphasy) [inline lib/models/film.js]: I think the `actors` property will be a list (array) of IDs. So, the object should be placed inside of an array.

```javascript
actors: [
  {
    type: Schema.Types.ObjectId,
    ref: 'Actor',
  }
]
```

File: lib/routes/actors.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+router
+    .get('/', (req, res, next) => {
+        // res.send('hello world')
+	Actor.find()
+            .then(actors => res.send(actors))
+            .catch(next);
+})
+
+    .get('/actor/:id', (req, res, next) => {
```
Comment (jluangphasy) [inline lib/routes/actors.js]: I suggest removing the `actor` in your path, so the URL doesn't become lengthy. Plus, your prefix you set in `app.js` for the resource already tells that you're working with the actors. When there's an ID provided, you can tell the request is for one single actor.

File: lib/models/actor.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const schema = new Schema({
```
Comment (GloriaAnholt) [inline lib/models/actor.js]: this should be named "Actor" since it's the actor schema.

File: test/api/films.test.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+	before(() => mongoose.connection.dropDatabase());
+
+	const request = chai.request(app);
+
+	const godzilla = {
+		title: 'Godzilla',
+		studio: '589a503f2fe3c376dc88b890',
+		released: '1954-11-03T00:00:00.000Z',
+		actors: '589a503f2fe3c376dc88b795',
```
Comment (jluangphasy) [inline test/api/films.test.js]: The `actors` should be an array of actor IDs. This goes for line 32 and 40.

**Pull request closed without merging**